src/test/java/guitests/PanelFocusTest.java

```diff
@@ -0,0 +1,49 @@
+package guitests;
+
+import org.junit.Test;
+
+import org.loadui.testfx.utils.FXTestUtils;
+import prefs.ConfigFileHandler;
+import prefs.GlobalConfig;
+import prefs.PanelInfo;
+import prefs.Preferences;
+import ui.issuepanel.PanelControl;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+
+public class PanelFocusTest extends UITest {
+
+    @Override
+    public void launchApp() { FXTestUtils.launchApp(TestUI.class, "--testconfig=true", "--bypasslogin=true"); }
+
+    @Override
+    public void setupMethod() {
+        ConfigFileHandler configFileHandler =
+                new ConfigFileHandler(Preferences.DIRECTORY, Preferences.TEST_CONFIG_FILE);
+        GlobalConfig globalConfig = new GlobalConfig();
+
+        PanelInfo test1 = new PanelInfo();
+        PanelInfo test2 = new PanelInfo();
+        PanelInfo test3 = new PanelInfo();
+        List<PanelInfo> panels = new ArrayList<>();
+        panels.add(test1);
+        panels.add(test2);
+        panels.add(test3);
+
+        globalConfig.setPanelInfo(panels);
+        configFileHandler.saveGlobalConfig(globalConfig);
+    }
+
+    @Test
+    public void panelFocusOnFirstPanelAtStartupTest() {
+        PanelControl panelControl = (PanelControl) find("#dummy/dummy_col0").getParent();
+
+        sleep(120000);
```

Comment:
Oops. It was not necessary, I was just debugging something. I will clean it
up.
On Sep 13, 2015 6:57 PM, "Darius Foo" <notifications@github.com> wrote:

> In src/test/java/guitests/PanelFocusTest.java
> <https://github.com/HubTurbo/HubTurbo/pull/956#discussion_r39346578>:
>
> > +        PanelInfo test2 = new PanelInfo();
> > +        PanelInfo test3 = new PanelInfo();
> > +        List<PanelInfo> panels = new ArrayList<>();
> > +        panels.add(test1);
> > +        panels.add(test2);
> > +        panels.add(test3);
> > +
> > +        globalConfig.setPanelInfo(panels);
> > +        configFileHandler.saveGlobalConfig(globalConfig);
> > +    }
> > +
> > +    @Test
> > +    public void panelFocusOnFirstPanelAtStartupTest() {
> > +        PanelControl panelControl = (PanelControl) find("#dummy/dummy_col0").getParent();
> > +
> > +        sleep(120000);
>
> The UI tests are run sequentially, so this will add 2 minutes to the
> entire thing... You can use UITest. waitUntilNodeAppears instead with
> some panel id.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/HubTurbo/HubTurbo/pull/956/files#r39346578>.
>